common/reportMetrics.py

```diff
@@ -369,108 +373,343 @@ def process_replay(file_path):
+                        next_line = log_file.readline()
+                        if next_line[:3] == 'ARG' or next_line[:4] == 'DATA' or next_line[:7] == 'MESSAGE':
+                            break
+                    log_file.seek(i)
+                else:
+                    extract_fields_db2(current_obj, line)
+
+            # if the last loop results in chunkLines being set...
+            if line_count == parameters['chunkLines']:
+                logger.debug("--- Chunk creation time: %s seconds ---" % (time.time() - start_time))
+                send_data(current_row, log_file_path, chunk_count)
+                current_row = []
+                chunk_count += 1
+                line_count = 0
+                start_time = time.time()
+
+            line = log_file.readline()
+        # last chunk
+        if len(current_row) != 0:
+            # build json entry
+            entry = dict()
+            entry['tag'] = 'db2-replay'
+            entry['eventId'] = str(current_obj.pop('timestamp'))
+            entry['data'] = current_obj
+            current_row.append(entry)
+            logger.debug("--- Chunk creation time: %s seconds ---" % (time.time() - start_time))
+            send_data(current_row, log_file_path, chunk_count)
+            chunk_count += 1
+        logger.debug("Total chunks created: " + str(chunk_count))
+    return
+
+
+def extract_fields_db2(obj, line):
+    # need to put spaces before :
+    matches = re.findall('[A-Z]+:', line)
+    for match in matches:
+        split_at = line.find(':', line.find(match))
+        line = line[:split_at] + ' ' + line[split_at:]
+
+    line_arr = line.split()
+    while len(line_arr) > 0:
+        rev_array = line_arr[::-1]
+        last_val_pos = len(line_arr) - rev_array.index(':')
+        last_val = ' '.join(line_arr[last_val_pos:])
+        last_key_pos = last_val_pos - 2
+        last_key = line_arr[last_key_pos]
+        obj[last_key] = last_val
+        if last_key_pos == 0:
+            break
+        line_arr = line_arr[:last_key_pos]
+    return
+
+
+def replay_gpfs(log_file_path):
+    print('Replaying gpfs file')
+    with open(log_file_path) as log_file:
+        line_count = 0
+        chunk_count = 0
+        current_row = []
+        start_time = time.time()
+        for line in log_file:
+            # skip empty lines
+            if not line.strip():
+                continue
+            # if the last loop results in chunkLines being set...
+            if line_count == parameters['chunkLines']:
+                logger.debug("--- Chunk creation time: %s seconds ---" % (time.time() - start_time))
+                send_data(current_row, log_file_path, chunk_count)
+                current_row = []
+                chunk_count += 1
+                line_count = 0
+                start_time = time.time()
+            # build json entry
+            entry = dict()
+            entry['tag'] = 'gpfs-replay'
```

Comment:
isn't there any host to get from the file? or can we just use localhost

Reply:
see updated tag. 

https://www.ibm.com/support/knowledgecenter/en/STXKQY_4.2.3/com.ibm.spectrum.scale.v4r23.doc/bl1pdg_errlogs.htm

> The GPFS log file is named mmfs.log.date.nodeName